Chat 35

Artemiy555 opened the pull request.

Artemiy555 committed:
add authFilter

Artemiy555 committed:
fix

Artemiy555 committed:
fix

Artemiy555 committed:
commit

Artemiy555 committed:
fix

Artemiy555 committed:
Merge branch 'develop' of https://github.com/zimajava/dev-ed-java-spec into CHAT-35

add log in Filter
refactoring properties

YaroslavSemeniuk commented on src/main/java/org/zipli/socknet/models/Chat.java:
```diff
@@ -10,6 +11,7 @@
 @Getter
 @Setter
 @Document
+@NoArgsConstructor
```
Валентин нам советовал создавать конструкторы вручную, а не при помощи Lombok чтобы не возникало проблем в будущем. Оставляем аннотацию?

Artemiy555 replied:
Если я правильно помню, он говорил о конструкторах с параметрами. А не пустых.

Valek-Z commented on src/main/java/org/zipli/socknet/repositories/modelsRepositories/UserRepository.java:
```diff
@@ -7,13 +7,16 @@
-    List<User> getByUserName(String userName);
+    User getUserByEmail(String email);
+    User getByUserName(String userName);
+    User findByUserName(String userName);
```
Зачем этот метод и чем отличается от getByUserName?
и ниже findByEmail == getUserByEmail ....

Valek-Z commented on src/main/java/org/zipli/socknet/repositories/modelsRepositories/UserRepository.java:
```diff
@@ -7,13 +7,16 @@
-    List<User> getUsersByConfirm(boolean isConfirm);
-    List<User> deleteByEmail(String email);
-    List<User> deleteByUserName(String userName);
+    List<User> getUserByConfirm(boolean isConfirm);
```
название getUser_s_ByConfirm
getAllConfirmUsers

Valek-Z commented on src/main/java/org/zipli/socknet/repositories/modelsRepositories/UserRepository.java:
```diff
@@ -7,13 +7,16 @@
-    List<User> deleteByUserName(String userName);
+    List<User> getUserByConfirm(boolean isConfirm);
+    void deleteByEmail(String email);
+    List<User> deleteAllByUserName(String userName);
```
Не понял кейс у нас может быть много пользователей с одним именем? 
тогда почему  :

> User getByUserName(String userName);

Valek-Z commented on src/main/java/org/zipli/socknet/security/jwt/AuthTokenFilter.java:
```diff
@@ -0,0 +1,55 @@
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    private final JwtUtils jwtUtils;
+
```
зачем пустая строка?

Valek-Z commented on src/main/java/org/zipli/socknet/security/services/UserDetailsServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+	private final UserRepository userRepository;
```
А зачем мы создавали  интерфейс  репозитория что бы не пользоваться?

Artemiy555 replied:
UserDetails userDetails = userDetailsService.loadUserByUsername(username);
Используем в фильтре

Valek-Z commented on src/test/java/org/zipli/socknet/security/jwt/AuthTokenFilterTest.java:
```diff
@@ -0,0 +1,59 @@
+    UserDetailsServiceImpl userDetailsService;
+
+    @Test
+    void doFilterInternalPass() {
```
А что мы в тесте проверяем?

Valek-Z commented on src/test/java/org/zipli/socknet/security/jwt/AuthTokenFilterTest.java:
```diff
@@ -0,0 +1,59 @@
+import java.io.IOException;
+
+@SpringBootTest
+class AuthTokenFilterTest {
```
А что мы в тестах проверяем?

Valek-Z commented on src/test/java/org/zipli/socknet/security/services/UserDetailsServiceImplTest.java:
```diff
@@ -0,0 +1,35 @@
+            "dsaaaaa");
+
+    @Test
+    void loadUserByUsername() {
```
Отказной тест где?

Valek-Z commented on src/test/java/org/zipli/socknet/security/services/UserDetailsServiceImplTest.java:
```diff
@@ -0,0 +1,35 @@
+class UserDetailsServiceImplTest {
+
+    @Autowired
+    UserRepository userRepository;
```
Для тестов сетапим тестовую базу!

Artemiy555 committed:
UserRepository.java: removed unnecessary methods
AuthTokenFilterTest.java: fix test
UserDetailsServiceImplTest.java: setup test base

Valek-Z commented on src/test/java/org/zipli/socknet/security/jwt/AuthTokenFilterTest.java:
```diff
@@ -0,0 +1,62 @@
+        try {
+            filter.doFilter(mockReq, mockResp, mockFilterChain);
+        } catch (ServletException | IOException e) {
+            assertNotEquals("",e.getMessage());
```
Это не проверка ожидаемого поведения.

Artemiy555 committed:
add AuthenticationException.java and test to it
fix AuthTokenFilter.java and test to it

Artemiy555 committed:
fix AuthTokenFilterTest
